### DISCUSSION TITLE
WEBootstrap theme (and themes in general)

### DISCUSSION THREAD

**Discussion by timschofield** [timschofield/webERP#290] [Development Discussion & Specification]
I think I would like to remove this theme for now. It looks like it might be good, but I can't get it working.

Any objections?
Tim

**Comment by dalers** [comment]
No objections, "Professional" is the only theme I use. Iirc I think I got the WEBootstrap theme once for login and it wasn't pretty.

_Edit 2025-11-23 now use Xenos as it "appears" to be the defacto standard_ 

On a related note, what is the purpose or benefit of each of the themes?

- WEBootstrap
- Aguapop
- Arc-dark
- Default
- Fluid
- Fresh
- Gel
- Professional
- Silverwolf
- Wood
- Xenos

Imho, it would be better to have fewer themes that are well defined and differentiated to a) make it easier for users to pick a suitable one, and b) to simplify testing and maintenance for the core team.

![image](https://github.com/user-attachments/assets/e890ea27-cad2-4a1a-b5c2-d08ed4b2fb2f)

**Comment by timschofield** [comment]
Basically we started out with the default theme, and then over the years
various people (including me) have contributed others. Modern thinking
would probably be that we just have Light/Dark which is how I did the web
site. The problem is I think people have their favourite.

On Mon, 9 Dec 2024 at 21:40, Dale Scott ***@***.***> wrote:

> No objections, "Professional" is the only theme I use. Iirc I think I
> ended up with WEBootstrap for the login theme and it wasn't pretty.
>
> On a related note, what is the purpose or benefit of each of the themes?
>
>    - WEBootstrap
>    - Aguapop
>    - Arc-dark
>    - Default
>    - Fluid
>    - Fresh
>    - Gel
>    - Professional
>    - Silverwolf
>    - Wood
>    - Xenos
>
> Imho, I would prefer to support fewer themes with well defined and
> differentiated purposes.
>
> image.png (view on web)
> <https://github.com/user-attachments/assets/e890ea27-cad2-4a1a-b5c2-d08ed4b2fb2f>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/290#discussioncomment-11513458>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LEGG2UMY4OOFYDNJ2L2EYE4TAVCNFSM6AAAAABTJW3VWGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNJRGM2DKOA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by andrewcouling** [comment]
No objections to removing WEBootstrap Theme.
A while ago, I worked on implementing a lightweight bootstrap css framework, but didnt get very far.

M own preference is Xenos.

:bulb: We could do another survey to understand preference for existing themes.

**Comment by pakricard** [comment]
Hi:
No objections deleting WEBootstrap (or mobile as I can't get it working properly too).

We use 3 themes, depending on the installation:
- default for production installation
- gel for staff training installation
- aguapop for development

This is a very visual way to know which webERP you are working in and prevent data mess (like staff training inserting rubbish data on the production DB, or real data being inserted into the training one and lost the next day).

It works better than a message on the header and footer warning the user, etc. I forced WWW_users.php to accept only one theme, depending on the usage installation, and users can't change their theme on UserSettings.php

Agree that we don't need 12 themes. With half of them we could please everybody and if someone requires another one, it's reasonably simple to change the colours to one of them and get a different one.

What would be really nice to make them 100% responsive, but I guess should be a post v5 work.

**Reply by dalers** [reply]
> This is a very visual way to know which webERP you are working in and prevent data mess...

Fwiw, I've used a commerical proprietary system that changed color if you were using the demo/test/sandbox database. Very useful!

> We use 3 themes, depending on the installation:
> - default for production
> - gel for staff training
> - aguapop for development

Is there anything about these themes _other than their color_ that makes one better than the others for each type of installation? Would you prefer if all three installations were **_identical except for color_**?

**Reply by pakricard** [reply]
> Is there anything about these themes _other than their color_ that makes one better than the others for each type of installation? Would you prefer if all three installations were **_identical except for color_**?

We prefer the modules' menu on the left (some themes have it below the header). Other than that, I don't care much. webERP is a business software, not TikTok ;-) so having a simple theme is best. 
Just selected these 3 ones (default,gel, aguapop) for the colour differences.
Anyway, I'm testing professional theme now, and will use it once I'm at v5. Looks cool and simple.

**Comment by timschofield** [comment]
I have removed the Webootstrap and mobile themes as that seemed to be consensus. They are still in the history so if in the future somebody wants to resurect them they can

**Comment by dalers** [comment]
I don't know if this means anything but NetBeans' PHP analyzer shows two issues in the css for arc-dark theme (and no issues for any of the others).

![image](https://github.com/user-attachments/assets/a5120c16-d292-44d8-8428-d9a442549931)

![image](https://github.com/user-attachments/assets/0685920a-dbc1-4763-b111-0c2cf9537517)

**Comment by timschofield** [comment]
I have fixed this Dale, thanks for the report

Tim

ps As I mentioned to Enrique earlier can we keep issues separate to
stop my head exploding :)


On Wed, 11 Dec 2024 at 22:32, Dale Scott ***@***.***> wrote:
>
> I don't know if this means anything but NetBeans' PHP analyzer shows two issues in the css for arc-dark theme (and no issues for any of the others).
>
> image.png (view on web)
>
> image.png (view on web)
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by hinjolicious** [comment]
I think the themes should be replaced with a neater one like these:

<img width="334" height="307" alt="login" src="https://github.com/user-attachments/assets/71222927-4647-48fe-8575-6576f8000fff" />

A floating login dialog


<img width="638" height="446" alt="zen" src="https://github.com/user-attachments/assets/451e53fa-aebb-473d-8a59-71dd18fe9d29" />

Zen theme


<img width="626" height="287" alt="woody" src="https://github.com/user-attachments/assets/7d549bf4-0d58-4624-8b65-d2ceb9cec604" />

The New Woody theme


<img width="630" height="497" alt="nemo" src="https://github.com/user-attachments/assets/6041c1fd-118d-498f-ba92-1edb81660c23" />

A fancy theme with animated schooling fishes! 😄 


But I'm using the old WebERP version, so adapting it to the current version woulb not be easy, also the UI structure might have to be changed a bit too!

**Reply by timschofield** [reply]
These themes look good. I think the css needs a complete overhaul, there is massive amounts of duplicated code in there (my fault mainly!) I think if we structure it better, thren making changes will be a lot easier.

Are you really still on 3.03??

Tim

**Reply by hinjolicious** [reply]
Yes, I've modified it along the way, but it's still functioning well for our needs.

I just visited WebERP again after a long time and I think we need to keep this gem going for a long time to come.

Let me see if I can contribute again to make the themes more appealing for our time.

Hinjo

**Reply by timschofield** [reply]
Would be great if you could contribute in some way!

It's very satisfying to hear that code we wrote all those years ago is still functioning!

Tim

**Reply by hinjolicious** [reply]
<img width="954" height="565" alt="Screenshot 2025-11-26 at 01-53-04 webERP - Main Menu" src="https://github.com/user-attachments/assets/4332bebd-df09-43b9-8db6-24f1d70e590a" />

"Zen Theme" for WebERP v4.15.2 👍 
Let me know what do you think about this cool and calm theme?

**Comment by dalers** [comment]
@hinjolicious I like it - but sorry, not enough to leave Xenos ;-)

Anyone's "best" theme is subject to many factors and opinions and concensus is unlikely. Instead of adding more themes to webERP, what about fewer "built-in" themes but have a repo where additional themes could be download and "installed" (perhaps including a couple screenshots that feature the theme).

**Reply by hinjolicious** [reply]
I agree, the included theme should be limited just several light and dark themes, preverably the most generic, nice and clean ones.
For other custom themes should be made available as an optional download.

**Reply by hinjolicious** [reply]
But, the current included themes is not there yet ;)
